recipe/meta.yaml

```diff
@@ -28,30 +28,31 @@ app:
 
 requirements:
   host:
-    - hatchling >=1.5.0
+    - hatchling >=1.21.1
     - hatch-jupyter-builder >=0.3.2
     - pip
     - python
   run:
     - python
     - async-lru >=1.0.0
+    - httpx >=0.25.0
+    - importlib_resources >=1.4   # [py<39]
     - importlib-metadata >=4.8.3  # [py<310]
-    - importlib_resources >=1.4  # [py<39]
-    - ipykernel
+    - ipykernel >=6.5.0
     - jinja2 >=3.0.3
     - jupyter_core
-    - jupyter-lsp >=2.0.0
     - jupyter_server >=2.4.0,<3
-    - jupyterlab_server >=2.19.0,<3
+    - jupyter-lsp >=2.0.0
+    - jupyterlab_server >=2.27.1,<3
     - notebook-shim >=0.2
     - packaging
-    - traitlets
+    - tomli >=1.2.2  # [py<311]
     - tornado >=6.2.0
```

Comment:
setuptools? https://github.com/jupyterlab/jupyterlab/blob/a0461257b780435e7d7e0c7044f111f616d4d719/pyproject.toml#L50

Reply:
Yeah, it's used here at runtime https://github.com/jupyterlab/jupyterlab/blob/v4.2.5/jupyterlab/federated_labextensions.py#L464. I'll leave a comment